xarray/core/duck_array_ops.py

```
@@ -325,6 +325,9 @@ def f(values, axis=None, skipna=None, **kwargs):
             nanname = "nan" + name
             func = getattr(nanops, nanname)
         else:
+            # Method 1
+            if name in ["sum", "prod"]:
```

should we always do this instead of restricting to just sum and prod? It seems to make sense for all other functions too (argmin, argmax, median, var, std, cumsum, cumprod)?

---

Only `prod` and `sum` take the `min_counts` argument. I think because

```python
da = xr.DataArray([np.NaN, np, np.NaN])
da.sum()
# returns 0 while
da.mean()
# returns np.NaN
´´´
Maybe it could be expanded to cumsum and cumprod. Adding `min_counts` to others could make sense but would be a larger undertaking.

---

If `min_count` is passed when it can't do affect the result, should we raise an error? (It's also possible I'm not understanding all the context here)

---

> If min_count is passed when it can't do affect the result, should we raise an error? 

This is the current behaviour AIUI. @mathuase wants `min_count` to be ignored in that case; which I agree is more convenient from the user-perspective